google/cloud/log.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/log.h"
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+static_assert(sizeof(Severity) <= sizeof(int),
+              "Expected Severity to fit in an integer");
+
+static_assert(static_cast<int>(Severity::LOWEST) <
+                  static_cast<int>(Severity::HIGHEST),
+              "Expect LOWEST severity to be smaller than HIGHEST severity");
+
+std::ostream& operator<<(std::ostream& os, Severity x) {
+  char const* names[] = {
+      "TRACE", "DEBUG",    "INFO",  "NOTICE", "WARNING",
```

Comment:
Won't clang-format remove the extra spacing used here for alignment?

Reply:
Actually clang-format *introduced* the extra space.  I am not complaining :smile: